services/painting-process-equipment-defect-detection-model-service/app/dependencies.py

```
@@ -0,0 +1,81 @@
+
+# 애플리케이션 시작 시 설정 파일, 모델 로드 및 explainer 재생성 함수
+async def load_resources(config_path: str = "app/models/model_config.yaml"):
+    global _model, _explainer, _config, _initialization_complete # 플래그를 global로 선언
+
+    try:
+        with open(config_path, 'r', encoding='utf-8') as f:
+            _config = yaml.safe_load(f)
+        print("설정 파일 로드 완료")
+    except FileNotFoundError:
+        print(f"오류: 설정 파일을 찾을 수 없습니다. 경로를 확인해주세요: {config_path}")
+        raise FileNotFoundError(f"Configuration file not found at {config_path}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)